presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java

```diff
@@ -226,4 +227,45 @@ public void testQueryOutputSizeExceeded()
             assertEquals(queryInfo.getErrorCode(), EXCEEDED_OUTPUT_SIZE_LIMIT.toErrorCode());
         }
     }
+
+    @Test
+    public void testQueryCountMetrics()
+            throws Exception
+    {
+        DispatchManager dispatchManager = queryRunner.getCoordinator().getDispatchManager();
+        // Create a total of 4 queries to test concurrency limit and ensure Queued Queries
+        createQueries(dispatchManager, 3);
+
+        // This is to ensure the 4th Query we want to monitor is Queued
+        while (dispatchManager.getStats().getRunningQueries() != 3) {
+            Thread.sleep(25);
+        }
+
+        QueryId queryId = dispatchManager.createQueryId();
```

Comment:
Thanks for updating the test. But this only tests 1 queued query. Do you think creating a few more of them would make the tests more meaningful? Something like
```
create r(=3) queries;
wait until all r queries are running;

create q queries;
assert q queries are queued;
while (dispatchManager.getStats().getQueuedQueries() + dispatchManager.getStats().getRunningQueries() > 0
                && stopwatch.elapsed().toMillis() < 60000) {
    numQueued = count queuedQueries from dispatchManager;
    assert numQueued == stats.queuedQueries;
    assert numQueued <= numQueued of last round 
 
    numRunning = count runningQueries from dispatchManager;
    assert numRunning == stats.runningQueries;
    assert numRunning <= numRunning from last round
    assert numRunning <= r;

    sleep(x); // x be a small number
}
```

Reply:
As discussed offline, I have updated the tests accordingly.